tests/sles4sap/ensa/netweaver_ensa2_web_methods.pm

```diff
@@ -14,14 +14,26 @@ use testapi;
 use hacluster;
 use lockapi;
 use serial_terminal qw(select_serial_terminal);
-
+use version_utils 'is_sle';
 
 sub webmethod_checks {
     my ($self, $instance_id) = @_;
+    my $outputs;
+    my $looptime = 300;
+
     # General status will help with troubleshooting
     $self->sap_show_status_info(cluster => 1, netweaver => 1, instance_id => $instance_id);
     record_info('ENSA check', "Executing 'HACheckConfig' and 'HACheckFailoverConfig'");
-    $self->sapcontrol(webmethod => 'HACheckConfig', instance_id => $instance_id);
+    while ($outputs = $self->sapcontrol(webmethod => 'HACheckConfig', instance_id => $instance_id, return_output => 1)) {
+        if ($outputs =~ /ERROR/) {
+            record_info("ERROR found in HACheckConfig: $outputs", "sleep 30s and try again");
+            sleep 30;
+            $looptime -= 30;
+            last if ($looptime <= 0);
+            next;
+        }
+        last;
```

Comment:
Maybe there's a Perl way to directly put all 3:
- run command 
- save the output to `$output` (do we really need to save it? Is it for failure debug purpose only right?
- check against `ERROR`

all of them within the `while`  condition. Doing so could allow to remove  this `last - next - last` logic.

Reply:
It is for failure debug purpose only.